cordformation/src/test/kotlin/net/corda/plugins/DockerformTest.kt

```diff
@@ -157,4 +157,87 @@ class DockerformTest : BaseformTest() {
         val bankOfCordaPortsList = bankOfCorda["ports"] as List<Any>
         assertThat(bankOfCordaPortsList).contains("22235:22235")
     }
+
+    @Suppress("unchecked_cast")
+    @Test
+    fun `deploy two nodes with Docker and external service`() {
+        val runner = getStandardGradleRunnerFor(
+                "DeployTwoNodeCordappWithExternalService.gradle",
+                "prepareDockerNodes")
+
+        val result = runner.build()
+        assertThat(result.task(":prepareDockerNodes")!!.outcome).isEqualTo(TaskOutcome.SUCCESS)
+
+        val dockerComposePath = getDockerCompose()
+
+        val yaml = dockerComposePath.toFile().bufferedReader().use { reader ->
+            Yaml().load(reader) as Map<String, Any>
+        }
+
+        val services = yaml["services"] as Map<String, Any>
+
+        val external = services["example-service"] as Map<String, Any>
+        assertThat(external).containsOnlyKeys("container_name","volumes","image","ports","expose","privileged","command","environment")
+
+        val name = external["container_name"] as String
+        assertThat(name).isEqualTo("example-service")
+
+        val image = external["image"] as String
+        assertThat(image).isEqualTo("docker.io/bitnami/java:latest")
+
+        val volumes = external["volumes"] as List<String>
+        assertThat(volumes).contains("C:\\Projects\\gs-rest-service\\target\\CordaDevTestAPI-0.0.1-SNAPSHOT.jar:/home/CordaDevTestAPI.jar")
+
+        val environment = external["environment"] as List<String>
+        assertThat(environment).contains("rpcUsername=user1")
+        assertThat(environment).contains("rpcPassword=pass2")
+
+        val ports = external["ports"] as List<String>
+        assertThat(ports).contains("8080:8080")
```

Comment:
This line shows twice `assertThat(ports).contains("8080:8080")`

Reply:
Not quite! one is for 8080, the other for 8000.